### PR TITLE
Slide deck with photos during stages of life

### LINKED ISSUE #9: Background Page/Blog Page Optional Tag
- [ ] Remove optional tag in index.html

### LINKED ISSUE #10: Background Page Create Enum
- [ ] Create enum -1, 1 for plusSlide in index.html

### LINKED ISSUE #12: Footer move to bottom of screen
Move footer to the bottom of screen on home page and slides page.

Screenshots:
https://screenshot.googleplex.com/utw32RbJ4OF.png
https://8080-5efc772e-925a-4fef-a0ac-77c0f89b1684.us-west1.cloudshell.dev/?authuser=0

### PR DESCRIPTION
**Pull request by jaysaleh**
This is the background page for my portfolio. This page includes a header for navigating between tabs as well as a slideshow with different pictures of me/locations at points throughout my life. script.js has the code for navigating between the slides.

Need to rename files after merging to avoid conflicts.

Screenshots;
https://screenshot.googleplex.com/pj7Agnfs5ZX.png
https://screenshot.googleplex.com/LA6khK0MgPi.png
https://screenshot.googleplex.com/cr3jyEEbSGb.png
https://screenshot.googleplex.com/vDjwDna6sif.png
https://screenshot.googleplex.com/MesB566sTth.png
https://screenshot.googleplex.com/JuKSd5Nro0y.png
https://screenshot.googleplex.com/RRTUPzTGNFk.png
https://screenshot.googleplex.com/5GGd3G4DOJT.png
https://screenshot.googleplex.com/RpCq1Fx6REZ.png
https://screenshot.googleplex.com/XF9uSE0GCrz.png

#4

### CODE REVIEW

**Review by f-camp — APPROVED** [review]

File: portfolio/src/main/webapp/index.html [hunk 1]
```diff
@@ -2,117 +2,119 @@
...
+        <img src="../images/grad.jpg" alt="Grad Picture" class="img-75">
+        <p>These are some of my closest friends in high school. It was a very nostalgic day. I miss the boys.</p>
+      </div>
+      <div class="mySlides fade">
+        <h1>First Day of College</h1>
+        <h3>Septemeber 21st 2018 - Stanford, CA</h1>
+        <img src="../images/stanford.jpg" alt="Stanford Picture" class="img-75">
+        <p>
+          I started school on Septemeber 21st 2018. Going into college I was premed. After taking CS106B, I found the content very enagaging 
+          and decided to switch majors to Computer Science. College was very intimiating at first with everyone saying it would be extremely
```
Comment (billie-cat) [inline portfolio/src/main/webapp/index.html]: nit: instead of the course code I would put the name so people know what the course is about.

File: portfolio/src/main/webapp/style.css [hunk 2]
```diff
@@ -26,83 +25,119 @@
...
+.dot-container {
+  text-align: center;
 }
 
-@-webkit-keyframes fade {
-  from {opacity: .4}
-  to {opacity: 1}
+.active, 
+.dot:hover {
+  background-color: #717171;
```
Comment (billie-cat) [inline portfolio/src/main/webapp/style.css]: Can the color value be 3 character hexadecimal notation? Here and elsewhere.

File: portfolio/src/main/webapp/style.css [hunk 3]
```diff
@@ -1,7 +1,6 @@
-* {
-    box-sizing:border-box;
-    font-family: Helvetica;
-
+* { 
+  box-sizing:border-box;
```
Comment (billie-cat) [inline portfolio/src/main/webapp/style.css]: There should be a space after the colon.

File: portfolio/src/main/webapp/index.html [hunk 4]
```diff
@@ -2,117 +2,119 @@
 <html>
   <head>
```
Comment (billie-cat) [inline portfolio/src/main/webapp/index.html]: I believe <head> and <body> tags can be omitted (maybe others). The style guide says: 

Omit optional tags (optional).

For file size optimization and scannability purposes, consider omitting optional tags. The HTML5 specification defines what tags can be omitted.

(This approach may require a grace period to be established as a wider guideline as it’s significantly different from what web developers are typically taught. For consistency and simplicity reasons it’s best served omitting all optional tags, not just a selection.)

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -48,10 +40,8 @@ function showSlides(n) {
   }
 
   for (i = 0; i < slides.length; i++) {
-      slides[i].style.display = "none";
-  }
-  for (i = 0; i < dots.length; i++) {
-      dots[i].className = dots[i].className.replace(" active", "");
+    slides[i].style.display = "none";
+    dots[i].className = dots[i].className.replace(" active", "");
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js]: I recommend adding parameter name comments. The style guide says:

“Parameter name” comments should be used whenever the value and method name do not sufficiently convey the meaning, and refactoring the method to be clearer is infeasible (see JS Practices advice). Their preferred format is before the value with "=":

someFunction(obviousParam, /* shouldRender= */ true, /* name= */ 'hello');

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -48,10 +40,8 @@ function showSlides(n) {
   }
 
   for (i = 0; i < slides.length; i++) {
-      slides[i].style.display = "none";
-  }
-  for (i = 0; i < dots.length; i++) {
-      dots[i].className = dots[i].className.replace(" active", "");
+    slides[i].style.display = "none";
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js]: I am assuming "none", "active", "block",  and "" have important meaning. I would create an enum to store all these variables. What do you think?

File: portfolio/src/main/webapp/script.js [hunk 7]
```diff
@@ -48,10 +40,8 @@ function showSlides(n) {
   }
 
   for (i = 0; i < slides.length; i++) {
-      slides[i].style.display = "none";
-  }
-  for (i = 0; i < dots.length; i++) {
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js]: Is it possible to use for-of? The style guide says:

With ES6, the language now has three different kinds of for loops. All may be used, though for-of loops should be preferred when possible.

File: portfolio/src/main/webapp/script.js [hunk 8]
```diff
@@ -12,28 +12,20 @@
...
 }
 
-// Dot control
+// Allows changing slides through the dots
 function currentSlide(n) {
   showSlides(slideIndex = n);
 }
 
+// Updated slides based on previous/next click or dot click
 function showSlides(n) {
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js]: Functions should have a JSDoc, so the comment format must be /** */. Example:

/**
 * Multiple lines of JSDoc text are written here,
 * wrapped normally.
 * @param {number} arg A number to do something to.
 */
function doSomething(arg) { … }

File: portfolio/src/main/webapp/index.html (hunk 1 above)
Comment (jaysaleh) [reply portfolio/src/main/webapp/index.html]: Added a link tag and brief description.

File: portfolio/src/main/webapp/style.css (hunk 2 above)
Comment (jaysaleh) [reply portfolio/src/main/webapp/style.css]: In this instance, since #717171 is not in #RRGGBB format it can't be condensed. I'll check over the rest of my hex declarations and make sure they are the shortest they can be.

File: portfolio/src/main/webapp/script.js (hunk 7 above)
Comment (jaysaleh) [reply portfolio/src/main/webapp/script.js]: I read more into for-of loops and it is possible, however, I would need  to use two for loops one for the dots and one for the slides. In this case, would it be better to loop once using a transitional `for` loop or twice using `for-of`?

Comment (billie-cat) [reply portfolio/src/main/webapp/script.js]: I am not a js expert, so I would say do what is easiest :)

File: portfolio/src/main/webapp/script.js (hunk 6 above)
Comment (jaysaleh) [reply portfolio/src/main/webapp/script.js]: I agree, added an enum storing all the variables with comments describing what each "state" means

File: portfolio/src/main/webapp/index.html (hunk 1 above)
Comment (billie-cat) [reply portfolio/src/main/webapp/index.html]: even better :)

File: portfolio/src/main/webapp/script.js [hunk 9]
```diff
@@ -12,39 +12,70 @@
...
 
+/** @param slideIndex slide currently being displayed */
 var slideIndex = 1;
+
+/**
+ * Supported class states for dots and slides.
+ * @enum {string}
+ */
+const states = {
+  SHOW_DOT: ' active',
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js]: is there a reason a whitespace is needed before the word "active"?

File: portfolio/src/main/webapp/script.js [hunk 10]
```diff
@@ -12,39 +12,70 @@
...
+ * specific slide. Function is called when either arrows are pressed.
+ * @param {integer} n where n = 1 to go forward or n = -1 to go back
+ */
 function plusSlides(n) {
   showSlides(slideIndex += n);
 }
 
-// Allows changing slides through the dots
+/**
+ * Function calls showSlides on the slide that corresponds to the dot pressed.
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js]: nit: this comment can cause confusion because the user does not know what showSlides means. I would rephrase to something like:

Allows changing of slides through the dots.

@param (integer} n where n is the number of the dot that is pressed

File: portfolio/src/main/webapp/script.js [hunk 11]
```diff
@@ -12,39 +12,70 @@
...
+  HIDE_DOT: '',
+  SHOW_SLIDE: 'block',
+  HIDE_SLIDE: 'none',
+}
+
 showSlides(slideIndex);
 
-// Next and previous controls
+/**
+ * Function computes which slide should be shown and class showSlides on that
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js]: Nit: this comment can also cause confusion because the user does not know what showSlides does. Instead of referring to other methods just explain what the method does. You could write:

Computes which slide should be show and then shows that specific slide.  Invoked when either arrows are pressed.
@param ...

File: portfolio/src/main/webapp/script.js [hunk 12]
```diff
@@ -12,39 +12,70 @@
...
+ * specific slide. Function is called when either arrows are pressed.
+ * @param {integer} n where n = 1 to go forward or n = -1 to go back
+ */
 function plusSlides(n) {
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js]: Optional: consider passing an enum where the only possible values are 1 and -1.

you could also add a TODO to do this in a separate change.

File: portfolio/src/main/webapp/script.js [hunk 13]
```diff
@@ -12,39 +12,70 @@
...
+ * Function calls showSlides on the slide that corresponds to the dot pressed.
+ * @param {integer} n where n is the number of the dot that is pressed
+ */
 function currentSlide(n) {
   showSlides(slideIndex = n);
 }
 
-// Updated slides based on previous/next click or dot click
+/**
+ * Function computes where the slide index is out of bounds and adjusts accordingly.
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js]: nit: no need to mention its a function as it is known from by line 29.

Comment (billie-cat) [inline portfolio/src/main/webapp/script.js]: "where the slide index" do you mean "when the slide index"?

File: portfolio/src/main/webapp/script.js [hunk 14]
```diff
@@ -12,39 +12,70 @@
...
+ * Function then sets all slides and dots to be not active and sets respective slide
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js]: Not sure from the comment what is the respective slide. Maybe consider rephrasing the comment?

**Review by billie-cat — COMMENTED** [review]
Please add the relevant issue to this pull request

File: portfolio/src/main/webapp/script.js (hunk 9 above)
Comment (jaysaleh) [reply portfolio/src/main/webapp/script.js]: The whitespace is necessary as I am attempting to add an attribute to the `"dot"` class turning it into `"dot active"`.

File: portfolio/src/main/webapp/script.js (hunk 14 above)
Comment (jaysaleh) [reply portfolio/src/main/webapp/script.js]: Added clarification which includes which slide and dot are changed by number

File: portfolio/src/main/webapp/script.js (hunk 12 above)
Comment (jaysaleh) [reply portfolio/src/main/webapp/script.js]: Added a TODO in index.html to see if its possible to add an enum.

File: portfolio/src/main/webapp/index.html (hunk 4 above)
Comment (jaysaleh) [reply portfolio/src/main/webapp/index.html]: Added TODO in index.html to remove optional tags in all .html files

File: portfolio/src/main/webapp/script.js (hunk 9 above)
Comment (billie-cat) [reply portfolio/src/main/webapp/script.js]: That makes sense. What I would do is either add a comment to describe this as it is not obvious from first looking at it. Or you could even make it an enum called ADD_ATTRIBUTE or create a const string called ADD_ATTRIBUTE and then SHOW_DOT: ADD_ATTRIBUTE + 'active'

**Comment by jaysaleh** [comment]
Adjusted comment and function name #7

File: portfolio/src/main/webapp/script.js (hunk 11 above)
Comment (jaysaleh) [reply portfolio/src/main/webapp/script.js]: Done

File: portfolio/src/main/webapp/script.js (hunk 13 above)
Comment (jaysaleh) [reply portfolio/src/main/webapp/script.js]: Done

Comment (jaysaleh) [reply portfolio/src/main/webapp/script.js]: fixed

File: portfolio/src/main/webapp/style.css (hunk 3 above)
Comment (jaysaleh) [reply portfolio/src/main/webapp/style.css]: fixed

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (jaysaleh) [reply portfolio/src/main/webapp/script.js]: done

File: portfolio/src/main/webapp/script.js (hunk 8 above)
Comment (jaysaleh) [reply portfolio/src/main/webapp/script.js]: Added comments

File: portfolio/src/main/webapp/script.js (hunk 9 above)
Comment (jaysaleh) [reply portfolio/src/main/webapp/script.js]: Done  #8

File: portfolio/src/main/webapp/script.js [hunk 15]
```diff
@@ -15,7 +15,7 @@
 /** @var slideIndex slide currently being displayed */
 var slideIndex = 1;
 
-/** @const ADD_ATTRIBUTE used to add a space in class attributes */
+/** @const ADD_ATTRIBUTE !string used to add a space in class attributes */
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js]: nit: The comment describes what it the code is doing not what its for/why it is needed. Does that make sense? I think you can write: used for adding an attribute. Or something of that nature. Not need to write adding a space.

Comment (jaysaleh) [reply portfolio/src/main/webapp/script.js]: Makes sense, fixed.

**Review by billie-cat — APPROVED** [review]

**Comment by jaysaleh** [comment]
#12

File: portfolio/src/main/webapp/index.html [hunk 16]
```diff
@@ -9,114 +9,116 @@
...
+            My mother birthed her second child (me) on August 12th, 2000 at 4:44 PM at Dameron Hospital in Stockton, California. 
+            I don't rememeber the journey but apparently I almost died. Shortly after, I traveled 16 miles north to my new crib. 
+            Above is a picture of me with all my new drip.
+          </p>
+        </div>
+        <div class="mySlides fade">
+          <h1>Brother's Ballpit Party</h1>
+          <h3>December 6th 2001 - Stockton, CA</h1>
+          <img src="../images/jay-ballpit.jpg" alt="Ballpit Picture" class="img-60">
+          <p>As you can see from the picture above, I was having a <b><i>blast.</i></b></p>
```
Comment (billie-cat) [inline portfolio/src/main/webapp/index.html]:
```suggestion
          <p>As you can see from the picture above, I was having a <b><i>blast</i></b>.</p>
```

File: portfolio/src/main/webapp/index.html [hunk 17]
```diff
@@ -9,114 +9,116 @@
...
+        </div>
+        <div class="mySlides fade">
+          <h1>First Zoo Trip</h1>
+          <h3>June 2002 - Lodi, CA</h1>
+          <img src="../images/jay-zoo.jpg" alt="Zoo Picture" class="img-60">
+          <p>
+            This was my first trip to Mickie Grove Park, a local park with a zoo. I was teriffied of animals before this day, but
```
Comment (billie-cat) [inline portfolio/src/main/webapp/index.html]:
```suggestion
            This was my first trip to Mickie Grove Park, a local park with a zoo. I was terrified of animals before this day, but 
```

File: portfolio/src/main/webapp/index.html [hunk 18]
```diff
@@ -9,114 +9,116 @@
...
+            This is where I went to Elementary School. I spent 2006 - 2011 here. As mentioned previously, I started school at 1st grade
+            so the majority of my time here was spent catching up. I'll take the next few slides to introduce my town/schools.
+            Shoutout to <br> Mr. McKilligan, you're a real one.
+          </p>
+        </div>
+        <div class="mySlides fade">
+          <h1>Millswood Middle School</h1>
+          <h3>August 2012 - May 2014 - Lodi, CA</h1>
+          <img src="../images/millswood.jpg" alt="Millswood Picture" class="img-80">
+          <p>This is where I went to middle school. I don't remember much from this time period so we'll just move on.</p>
```
Comment (billie-cat) [inline portfolio/src/main/webapp/index.html]:
```suggestion
          <p>This is where I went to Middle School. I don't remember much from this time period so we'll just move on.</p>
```

File: portfolio/src/main/webapp/index.html [hunk 19]
```diff
@@ -9,114 +9,116 @@
...
+        </div>
+        <div class="mySlides fade">
+          <h1>Lodi High School</h1>
+          <h3>August 2014 - May 2018 - Lodi, CA</h1>
+          <img src="../images/lodihigh.jpg" alt="Lodi High Picture" class="img-60">
+          <p>
+            This is my high school. I would say about 80% of my memories come from this place. I made a lot of freinds here through
```
Comment (billie-cat) [inline portfolio/src/main/webapp/index.html]:
```suggestion
            This is my High School. I would say about 80% of my memories come from this place. I made a lot of friends here through 
```

File: portfolio/src/main/webapp/index.html [hunk 20]
```diff
@@ -9,114 +9,116 @@
...
+          <p>This is where I went to middle school. I don't remember much from this time period so we'll just move on.</p>
+        </div>
+        <div class="mySlides fade">
+          <h1>Lodi High School</h1>
+          <h3>August 2014 - May 2018 - Lodi, CA</h1>
+          <img src="../images/lodihigh.jpg" alt="Lodi High Picture" class="img-60">
+          <p>
+            This is my high school. I would say about 80% of my memories come from this place. I made a lot of freinds here through 
+            several activities such as cross country, track and field, rocketry club, the 3D printing lab, and Science Olympiad. As 
+            much as I hated this place while I was there, I miss the memories I made with my friends before we all went our respective paths.
```
Comment (billie-cat) [inline portfolio/src/main/webapp/index.html]:
```suggestion
            much as I hated this place while I was there, I miss the memories I made with my friends before we all went our own respective paths.
```

File: portfolio/src/main/webapp/index.html [hunk 21]
```diff
@@ -9,114 +9,116 @@
...
+          </p>
+        </div>
+        <div class="mySlides fade">
+          <h1>High School Graduation</h1>
+          <h3>May 31st 2018 - Stockton, CA</h1>
+          <img src="../images/grad.jpg" alt="Grad Picture" class="img-75">
+          <p>These are some of my closest friends in high school. It was a very nostalgic day. I miss the boys.</p>
```
Comment (billie-cat) [inline portfolio/src/main/webapp/index.html]:
```suggestion
          <p>These are some of my closest friends in High School. It was a very nostalgic day. I miss the boys.</p>
```

File: portfolio/src/main/webapp/index.html [hunk 22]
```diff
@@ -9,114 +9,116 @@
...
+            I started school on Septemeber 21st 2018. Going into college I was premed. After taking <b><a href="https://web.stanford.edu/class/cs106b/" target="_blank" id="cs-link">CS106B</a></b> (Programing Abstractions), I found the content very enagaging 
+            and decided to switch majors to Computer Science. College was very intimiating at first with everyone saying it would be extremely 
+            diffcult. Looking back to my first day, I would tell myself to take it easy.
+          </p>
+        </div>
+        <div class="mySlides fade">
+          <h1>First Day at Google</h1>
+          <h3>June 15th 2020 - Lodi, CA</h1>
+          <img src="../images/google.jpg" alt="Google Picture" class="img-50">
+          <p>I'm super exicted to be a part of Google and for all the explorations I take this summer!</p>
```
Comment (billie-cat) [inline portfolio/src/main/webapp/index.html]: Can you rephrase: for all the explorations I take this summer? Not sure what you mean.

**Comment by billie-cat** [comment]
Here too, you should change the title of this pull request to describe what you have changed. 

"Try to keep your first line short, focused, and to the point. The clarity and utility to the reader should be the top concern.

By tradition, the first line of a CL description is a complete sentence, written as though it were an order (an imperative sentence). For example, say "Delete the FizzBuzz RPC and replace it with the new system.” instead of "Deleting the FizzBuzz RPC and replacing it with the new system.” You don’t have to write the rest of the description as an imperative sentence, though."

https://google.github.io/eng-practices/review/developer/cl-descriptions.html

File: portfolio/src/main/webapp/index.html (hunk 16 above)
Comment (jaysaleh) [reply portfolio/src/main/webapp/index.html]: done

File: portfolio/src/main/webapp/index.html (hunk 17 above)
Comment (jaysaleh) [reply portfolio/src/main/webapp/index.html]: done

File: portfolio/src/main/webapp/index.html (hunk 18 above)
Comment (jaysaleh) [reply portfolio/src/main/webapp/index.html]: done

File: portfolio/src/main/webapp/index.html (hunk 19 above)
Comment (jaysaleh) [reply portfolio/src/main/webapp/index.html]: done
